Fixed formatting/Added light&dark themes

vaitaitisr opened the pull request:
Fixed formatting on user summary page photo, added colors to the buttons in main page and changed About page's background

absplov commented:
If you only plan to create dark and light themes, instead of using DynamicResource you could use AppThemeBinding to set any property according to user system's dark theme settings. For example:

```
BackgroundColor="{AppThemeBinding Light={StaticResource LightBackgroundColor}, Dark={StaticResource DarkBackgroundColor}}"
```

Also, I would suggest to create an implicit style in Application.Resources to avoid adding the code above to every page:

```
<Application.Resources>
    <Color x:Key="LightBackgroundColor">White</Color>
    <Color x:Key="DarkBackgroundColor">Black</Color>

    <Style ApplyToDerivedTypes="True" TargetType="ContentPage">
        <Setter Property="BackgroundColor" Value="{AppThemeBinding Light={StaticResource LightBackgroundColor}, Dark={StaticResource DarkBackgroundColor}}"/>
    </Style>
</Application.Resources>
```

More info on [system light/dark themes](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/system-theme-changes) and [XAML styles](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/styles/xaml).

vaitaitisr commented:
Did some changes regarding AppThemeBinding;
Added button,label,searchbar and page style;
Changed up all pages to reference styles;
Fixed issue #14, I believe
![image](https://user-images.githubusercontent.com/112924297/193468692-7310a1b8-276f-42cd-8072-9b0ebdf266c3.png)

ArmandJas commented:
![image_2022-10-03_165746854](https://user-images.githubusercontent.com/95615520/193595952-304e2405-5f4f-4800-a946-7f77d4242ea1.png)
A minor detail, but the user details page's gender text box ends up blocking part of the icon, maybe try making it transparent?

ArmandJas commented on MySaver/Views/UserDetailsPage.xaml:
```diff
@@ -31,16 +27,12 @@
-                                  TextColor="Black"/>
+                            <Span Text="Gender" TextColor="Gray"/>
+                            <Span Text="&#x0a;" TextColor="Gray"/>
+                            <Span Text="Male" TextColor="Gray"/>
```
The text color here is explicitly gray, which overrides the light/dark theme settings, I think it can be a bit difficult to read gray text when using the dark theme so consider removing "TextColor="Gray" from label text. (also in the user summary page maybe)

vaitaitisr replied:
Changed it, although switching between light/dark themes requires to leave and come back to the same page to see the results. Gonna work on that a bit later

vaitaitisr commented on MySaver/Views/ListPage.xaml:
```diff
@@ -10,11 +10,15 @@
-            TextChanged="OnTextChanged" />
+            TextChanged="OnTextChanged" 
+            Style="{StaticResource Search}"/>
+            HorizontalOptions="Center" />
```
Missed that while resolving conflicts, my bad!

vaitaitisr commented:
> ![image_2022-10-03_165746854](https://user-images.githubusercontent.com/95615520/193595952-304e2405-5f4f-4800-a946-7f77d4242ea1.png) A minor detail, but the user details page's gender text box ends up blocking part of the icon, maybe try making it transparent?

On android everything looks good, still fixed it on windows machine. Forgot to check it out on windows machine the last time

ArmandJas commented:
Also, you should delete ratings.txt as it's deprecated.
